parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnReaderImpl.java

```diff
@@ -546,6 +573,11 @@ private void initDataReader(Encoding dataEncoding, byte[] bytes, int offset, int
     } catch (IOException e) {
       throw new ParquetDecodingException("could not read page in col " + path, e);
     }
+
+    if (dataEncoding.requiresSequentialReads(writerVersion) && previousReader != null) {
```

Comment:
Is this code only for the particular case of fixing the DeltaByteArrayReader bug?

If it is, then is the generalization of the RequiresPreviousReader necessary?